File: pset6/readability/readability.py
def main():
    text = input("Text: ")
    text = text.strip()
    letters = 0
    spaces = 0
    sentences = 0
    for character in text:
        if character.isalpha():
            letters += 1
        if character.isspace():
            spaces += 1
        if character == "." or character == "!" or character == "?":
            sentences += 1
    words = spaces + 1
    L = letters * 100 / words
    S = sentences * 100 / words
    result = round(0.0588 * L - 0.296 * S - 15.8)
    if(result >= 16):
        print(f"Grade 16+")
        return 0
    elif(result < 1):
        print(f"Before Grade 1")
        return 0
    print(f"Grade {result}")


if __name__ == "__main__":
    main()

File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>
#include <string.h>
#include <ctype.h>
//readability

int main(void)
{
    const string input = get_string("Text: ");
    const int n = strlen(input);
    int letters = 0;
    int words = 0;
    int sentences = 0;
    for (int i = 0; i < n; i++)
    {
        char actualChar = input[i];
        if (isalpha(actualChar))
        {
            letters++;
        }
        if (isspace(actualChar))
        {
            words++;
        }
        else
        {
            if (i == n - 1)
            {
                words++;
            }
        }



        if ((actualChar == '!') || (actualChar == '?') || (actualChar == '.'))
        {
            sentences++;

        }
        else
        {
            if (i == n - 1 && ((actualChar == '!') || (actualChar == '?') || (actualChar == '.')))
            {
                sentences++;
            }
        }
    }

    float L1 = (100 * letters) / (float)words;
    float S1 = (100 * sentences) / (float)words;

    float L = floorf(L1 * 100) / 100;
    float S = floorf(S1 * 100) / 100;


    int result = round(0.0588 * L - 0.296 * S - 15.8);
    if (result < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (result > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", result);
    }
}

